### PR TITLE
Fix race condition in cypress test by preventing typing in input before team loaded

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description: What code changed, and why?

If you have a team in the store already, it will populate the team save input before the team is fetched via the API. Since we didn't prevent typing in the input before the team was fetched via the API, cypress would find the input, find that the team name was correct (from the store cache), and then type. However, this would happen so fast the team wasn't fetched yet. When the fetch completed, it overwrote the updated team name cypress typed before save was clicked.

I've added functionality to disable the team name input before the team is fetched and while a save is in progress. The cypress tests should now pass since they are asserting that the input is enabled, preventing the race condition.

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

Also added base styling for the chefInput directive when disabled. It looks like this:

Screen Shot 2019-08-20 at 3 39 07 PM

### CODE REVIEW

File: e2e/cypress/integration/ui/settings/04_team_details.spec.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-describe('team management', () => {
```
Comment (tylercloke) [inline e2e/cypress/integration/ui/settings/04_team_details.spec.ts]: O L D C O P Y P A S T A (bad Tyler)

File: components/automate-ui/src/app/pages/team-details/team-details.component.ts [hunk 2]
```diff
@@ -95,7 +95,8 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
     };
     this.updateNameForm = fb.group({
       // Must stay in sync with error checks in team-details.component.html
-      name: ['Loading...', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]]
+      name: new FormControl({value: 'Loading...', disabled: true},
```
Comment (bcmdarroch) [inline components/automate-ui/src/app/pages/team-details/team-details.component.ts]: maybe a comment here about why we disable it to start

File: components/automate-ui/src/app/pages/team-details/team-details.component.ts [hunk 3]
```diff
@@ -279,6 +284,7 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
   public saveTeam(): void {
     this.saveSuccessful = false;
     this.saving = true;
+    this.updateNameForm.controls['name'].disable();
```
Comment (msorens) [inline components/automate-ui/src/app/pages/team-details/team-details.component.ts]: Couldn't you add `[disabled]="saving"` in the template so the disable/enable will be done automatically? (Notice that you are setting `this.saving` exactly where you are calling `disable()` and `enable()` in this function.)

Comment (tylercloke) [reply components/automate-ui/src/app/pages/team-details/team-details.component.ts]: Nope, if you try to use `[disabled]` with an input that is using formbuilder the compiler yells at you to not do that.

Comment (tylercloke) [reply components/automate-ui/src/app/pages/team-details/team-details.component.ts]: Really wish you could. I wanted to do it the way you described. Maybe we can. Need to look deeper but it appears formbuilder gets mad.

**Pull request merged**